src/Connectors/PhpRedisConnector.php

```diff
@@ -0,0 +1,260 @@
+        $clientOpts = array_merge($options, Arr::pull($servers, 'options', [ ]));
+
+        // Extract the array of Sentinel connection options from the rest of
+        // the client options
+        $sentinelOpts = array_intersect_key($clientOpts, $this->sentinelKeys);
+
+        // Filter the Sentinel connection options elements from the client
+        // options array
+        $clientOpts = array_diff_key($clientOpts, $this->sentinelKeys);
+
+        // Create a client by calling the Sentinel servers
+        $connector = function () use ($options) {
+            return $this->createClientWithSentinel($options);
+        };
+
+        // Create a connection and retry if this fails.
+        $connection = $this->retryOnFailure(function () use ($connector) {
+            return $connector();
+        });
+
+        return new PhpRedisConnection($connection, $connector, $sentinelOpts);
+    }
+
+    /**
+     * Create the Redis client instance
+     *
+     * @param  array  $options
+     * @return Redis
+     *
+     * @throws LogicException
+     */
+    protected function createClientWithSentinel(array $options)
+    {
+        $servers = $this->servers;
+        $service = isset($options['service']) ? $options['service'] : 'mymaster';
+        $timeout = isset($options['sentinel_timeout']) ? $options['sentinel_timeout'] : 0;
+        $persistent = isset($options['sentinel_peristent']) ? $options['sentinel_peristent'] : null;
+        $retryWait = isset($options['retry_wait']) ? $options['retry_wait'] : 0;
+        $readTimeout = isset($options['sentinel_read_timeout']) ? $options['sentinel_read_timeout'] : 0;
+        $parameters = isset($options['parameters']) ? $options['parameters'] : [];
+
+        // Shuffle the servers to perform some loadbalancing.
+        shuffle($servers);
+
+        // Check if the redis extension is enabled.
+        if (! extension_loaded('redis')) {
+            throw new LogicException('Please make sure the PHP Redis extension is installed and enabled.');
+        }
+
+        // Check if the extension is up to date and contains RedisSentinel.
+        if (! class_exists(RedisSentinel::class)) {
+            throw new LogicException('Please make sure the PHP Redis extension is up to date.');
+        }
+
+        // Try to connect to any of the servers.
+        foreach ($servers as $idx => $server) {
+            $host = isset($server['host']) ? $server['host'] : 'localhost';
+            $port = isset($server['port']) ? $server['port'] : 26739;
+
+            // Create a connection to the Sentinel instance.
+            $sentinel = new RedisSentinel($host, $port, $timeout, $persistent, $retryWait, $readTimeout);
+
+            try {
+                // Check if the Sentinel server list needs to be updated.
+                // Put the current server and the other sentinels in the server list.
+                $updateSentinels = isset($options['update_sentinels']) ? $options['update_sentinels'] : false;
```

Comment:
Why is this assignment not at the beginning of the method, like all other assignments? Would be a minor performance improvement as well.